packages/babel-parser/src/plugins/jsx/index.js

```diff
@@ -107,6 +107,15 @@ export default (superClass: Class<Parser>): Class<Parser> =>
             chunkStart = this.state.pos;
             break;
 
+          case charCodes.greaterThan:
+          case charCodes.rightCurlyBrace:
+            throw this.raise(
+              this.state.pos,
+              `Unexpected token \`${
+                this.input[this.state.pos]
+              }\`. Did you mean \`{'${this.input[this.state.pos]}'}\`?`,
```

Comment:
Nit: we should encourage the HTML entities `&rbrace;` and `&gt;`, which shouldn't require the separate `JSXExpressionContainer`.

Reply:
Would you be okay with the message suggesting both? i.e.
``Unexpected token `>`. Did you mean `{'>'}`, or `&gt;`?``

Personally I don't like seeing html entity codes in JSX.

Reply:
Sure.